bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerService.kt

```diff
@@ -128,12 +128,17 @@ class BusinessPartnerService(
         val partnerToUpsert = existingPartner ?: BusinessPartnerDb.createEmpty(upsertData.sharingState, upsertData.stage)
 
         val hasChanges = changeType == ChangelogType.CREATE || compareUtil.hasChanges(upsertData, partnerToUpsert)
+        val shouldUpdate = when {
+            upsertData.externalSequenceTimestamp == null -> true
+            existingPartner?.externalSequenceTimestamp == null -> true
+            else -> upsertData.externalSequenceTimestamp!!.isAfter(existingPartner.externalSequenceTimestamp)
+        }
 
-        if (hasChanges) {
-            changelogRepository.save(ChangelogEntryDb(sharingState.externalId, sharingState.tenantBpnl, changeType, stage))
+        if (hasChanges && shouldUpdate) {
+                changelogRepository.save(ChangelogEntryDb(sharingState.externalId, sharingState.tenantBpnl, changeType, stage))
 
-            copyUtil.copyValues(upsertData, partnerToUpsert)
-            businessPartnerRepository.save(partnerToUpsert)
+                copyUtil.copyValues(upsertData, partnerToUpsert)
+                businessPartnerRepository.save(partnerToUpsert)
         }
 
         return UpsertResult(hasChanges, changeType, partnerToUpsert)
```

Comment:
The upsert result should give the information whether the record did indeed change. Before there was only `hasChanges` as a criterion but now we have also `shouldUpdate`. In the UpsertResult we should therefore pass the result of both values.

Reply:
1. extend a new attribute to UpsertResult
2. return UpsertResult(hasChanges, **shouldUpdate**, changeType, partnerToUpsert)
3. .takeIf { it.hadChanges || **it.shouldUpdate** || sharingState.sharingStateType == SharingStateType.Error }